### ISSUE TITLE
[Feature Request] [stdlib] Add `argsort` to the builtin `sort` module

### ISSUE THREAD

**Issue by martinvuyk** [modularml/mojo#2979]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

Add argsort similar to [numpy's](https://numpy.org/doc/stable/reference/generated/numpy.argsort.html)

### What is your motivation for this change?

sorting inplace is not an option if you need to sort multiple arrays keeping the same permutations

### Any other details?

I don't know what would be needed in terms of `Span`'s capabilities. Because that permutation should be able to affect other items.
```mojo
var my_list_distances = List[UInt64]()
var my_structures = List[MyStruct]()
...
var permutation = argsort(my_list_distances)
var ordered_structs = my_structures[permutation]
```
Or for now we can just return a list of indexes

**Comment by JoeLoser** [comment]
I agree we should provide some facilities to support sorting things beyond just in-place.  This isn't currently a priority for the team, so I've put this as Postponed.  We'll share more about our Q3 priorities soon to give some insights into how the community and us can build toward those goals together.
